packages/client/test/rpc/eth/getFeeHistory.spec.ts

```diff
@@ -0,0 +1,161 @@
+  const { vm } = execution
+  const parentBlock = await chain.getCanonicalHeadBlock()
+  const vmCopy = await vm.copy()
+  // Set block's gas used to max
+  const blockBuilder = await vmCopy.buildBlock({
+    parentBlock,
+    headerData: {
+      timestamp: parentBlock.header.timestamp + BigInt(1),
+      gasUsed: bigIntToHex(gasUsed),
+    },
+    blockOpts: {
+      calcDifficultyFromHeader: parentBlock.header,
+      putBlockIntoBlockchain: false,
+    },
+  })
+  blockBuilder.gasUsed = gasUsed
+
+  const block = await blockBuilder.build()
+  await chain.putBlocks([block], false)
+  await execution.run()
+}
+
+tape(`${method}: should return 12.5% increased baseFee if parent block is full`, async (t) => {
+  const { chain, server, execution } = await setupChain(gethGenesisStartLondon(pow), 'powLondon')
+  const gasUsed = bytesToBigInt(hexToBytes(pow.gasLimit))
+
+  await produceFakeGasUsedBlock(execution, chain, gasUsed)
+  const req = params(method, ['0x2', 'latest', []])
+
+  const expectRes = (res: any) => {
+    const [, previousBaseFee, nextBaseFee] = res.body.result.baseFeePerGas as [
+      string,
+      string,
+      string
+    ]
+    const increase =
+      Number(
+        (1000n *
+          (bytesToBigInt(hexToBytes(nextBaseFee)) - bytesToBigInt(hexToBytes(previousBaseFee)))) /
+          bytesToBigInt(hexToBytes(previousBaseFee))
+      ) / 1000
+
+    t.equal(increase, 0.125)
+  }
+
+  await baseRequest(t, server, req, 200, expectRes)
+})
+
+tape(`${method}: should return 12.5% decreased base fee if the block is empty`, async (t) => {
+  const { chain, server, execution } = await setupChain(gethGenesisStartLondon(pow), 'powLondon')
+  const gasUsed = BigInt(0)
+
+  await produceFakeGasUsedBlock(execution, chain, gasUsed)
+  const req = params(method, ['0x2', 'latest', []])
+
+  const expectRes = (res: any) => {
+    const [, previousBaseFee, nextBaseFee] = res.body.result.baseFeePerGas as [
+      string,
+      string,
+      string
+    ]
+    const increase =
+      Number(
+        (1000n *
+          (bytesToBigInt(hexToBytes(nextBaseFee)) - bytesToBigInt(hexToBytes(previousBaseFee)))) /
+          bytesToBigInt(hexToBytes(previousBaseFee))
+      ) / 1000
+
+    t.equal(increase, -0.125)
+  }
+
+  await baseRequest(t, server, req, 200, expectRes)
+})
+
+tape(
+  `${method}: should return initial base fee if the block number is london hard fork`,
+  async (t) => {
+    const initialBaseFee = 1000000000n
+    const { server } = await setupChain(gethGenesisStartLondon(pow), 'powLondon')
+
+    const req = params(method, ['0x1', 'latest', []])
+
+    const expectRes = (res: any) => {
+      const [baseFee] = res.body.result.baseFeePerGas as [string]
+
+      t.equal(bytesToBigInt(hexToBytes(baseFee)), initialBaseFee)
+    }
+
+    await baseRequest(t, server, req, 200, expectRes)
+  }
+)
+
+tape(`${method}: should return 0x0 for base fees requested before eip-1559`, async (t) => {
```

Comment:
Not super sure about the `eth_feeHistory` spec, does this only work for 1559 chains, or also for non-1559 chains..?

Reply:
If by non-1559 chains you mean backward compatibility, then yes. The `eth_feeHistory` is fully backward compatible as per spec. 

`Legacy Ethereum transactions will still work and be included in blocks, but they will not benefit directly from the new pricing system. This is due to the fact that upgrading from legacy transactions to new transactions results in the legacy transaction’s gas_price entirely being consumed either by the base_fee_per_gas and the priority_fee_per_gas.`

Source: https://eips.ethereum.org/EIPS/eip-1559